lib/pbench/agent/results.py

```diff
@@ -102,100 +108,123 @@ def make_result_tb(self):
+        """
         tarball_md5 = Path(f"{tarball}.md5")
         try:
             hash_md5 = md5sum(tarball)
             tarball_md5.write_text(f"{hash_md5} {tarball.name}\n")
-        except Exception:
+        except Exception as exc:
             self.logger.error("md5sum failed for {}, skipping", tarball)
             try:
                 tarball.unlink()
             except OSError as exc:
                 if exc.errno != errno.ENOENT:
-                    self.logger.error("error removing failed tar ball, {}", tarball)
+                    self.logger.error("Error removing failed tarball, {}", tarball)
             try:
                 tarball_md5.unlink()
             except OSError as exc:
                 if exc.errno != errno.ENOENT:
                     self.logger.error(
-                        "error removing failed tar ball MD5, {}", tarball_md5
+                        "Error removing failed tarball MD5, {}", tarball_md5
                     )
-            sys.exit(1)
+            raise RuntimeError(f"Error in making tarball.md5, ({exc})") from exc
 
 
 class CopyResultTb:
+    """CopyResultTb - Copies tarball from agent to server with the help of RestAPI PUT request
+    """
+
     chunk_size = 4096
 
     def __init__(self, controller, tarball, config, logger):
+        """__init__ - Initializes some required attributes
+
+            Args
+                tarball -- path of tarball
+                tarball_md5 -- path of tarball's .md5
+                server_rest_url -- base uri
+                upload_url -- url where tarball needs to be send
+        """
         self.tarball = Path(tarball)
-        if not self.tarball.exists():
-            logger.error("tarball does not exist, '{}'", tarball)
-            sys.exit(1)
         self.tarball_md5 = Path(f"{tarball}.md5")
-        if not self.tarball_md5.exists():
-            logger.error("tarball's .md5 does not exist, '{}'", self.tarball_md5)
-            sys.exit(1)
+        self.check_tb_tbmd5()
         self.logger = logger
-        server_rest_url = config.results.get("server_rest_url")
+        server_rest_url = config.get("results", "server_rest_url")
         self.upload_url = f"{server_rest_url}/upload/ctrl/{controller}"
 
+    def check_tb_tbmd5(self):
+        """check_tb_tbmd5 - checks the existence of tb and tarball's .md5 file."""
+        if not self.tarball.exists():
+            raise FileNotFoundError(f"Tarball does not exist, '{self.tarball}'")
+        if not self.tarball_md5.exists():
+            raise FileNotFoundError(
+                f"Tarball's .md5 does not exist, '{self.tarball_md5}'"
+            )
+
     def read_in_chunks(self, file_object):
         data = file_object.read(self.chunk_size)
         while data:
             yield data
             data = file_object.read(self.chunk_size)
 
-    def copy_result_tb(self):
+    def copy_result_tb(self, token):
+        """copy_result_tb - copies tb from agent to given upload URL
```

Comment:
"given" upload URL implies that it's a parameter, so "to configured server upload URL" would probably be better.

Reply:
ok sure :)